main/test/ca/uwaterloo/flix/language/phase/TestParser.scala

```diff
@@ -1456,27 +1456,27 @@ class TestParser extends FunSuite with TestUtils {
 
   test("Expression.MatchLambda.05") {
     val input = "def f: Option[Int] -> Int = match None -> 42"
-    run(input, core = false)
+    expectError[NonExhaustiveMatchError](new Flix().addStr(input).compile())
```

Comment:
I think we should warn for lambda matches too, if they are not exhaustive. 

You would still be able to use LambdaMatch for both singleton enums (common) and tuples (common)